src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs

```diff
@@ -55,9 +59,10 @@ impl<'tcx> BorrowExplanation<'tcx> {
             },
             BorrowExplanation::UsedLaterInLoop(later_use_kind, var_or_use_span) => {
                 let message = borrow_desc + match later_use_kind {
-                    LaterUseKind::ClosureCapture => {
-                        "borrow captured here by closure, in later iteration of loop"
-                    },
+                    LaterUseKind::TraitCapture =>
+                        "borrow later captured here by trait object, in later iteration of loop",
```

Comment:
Pre-existing but: there... has to be a better way than having two copies of each message... :)

Why, for some messages, do we have not say "in later iteration of loop", anyway? (e.g., `FakeLetRead`)

Reply:
I think the two copies of each message is kind-of necessary, we can't just have one base message and append the "in later iteration of loop" because we remove the "later" from the earlier part of the message. We _could_ do that, but it'd be a lot of stiching together strings, so I think it's probably cleaner to not do that.